src/Controller/User/RegistrationController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Controller\User;
+
+use App\Entity\User;
+use App\Form\RegistrationForm;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\Routing\Attribute\Route;
+
+#[Route('/register', name: 'register')]
+final class RegistrationController extends AbstractController
+{
+    public function __invoke(
+        Request $request,
+        UserPasswordHasherInterface $passwordHasher,
+        EntityManagerInterface $em,
+        Security $security,
+    ) {
+        $user = new User();
+        $form = $this->createForm(RegistrationForm::class, $user);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var string $plainPassword */
+            $plainPassword = $form->get('password')->getData();
+
+            // encode the plain password
+            $user->setPassword($passwordHasher->hashPassword($user, $plainPassword));
```

Comment:
In our UserBundle, I'd really like to hash automatically. These one-off hashes introduce complexity & feel sloppy

Reply:
You mean like how fosuser bundle did? Doctrine event listener and eraseCredentials? Fyi eraseCredentials is deprecated as of 7.3